client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx

```diff
@@ -204,7 +213,23 @@ const GoalsStep: StepType = ( { navigation, flow } ) => {
 				<Step.CenteredColumnLayout
 					columnWidth={ 6 }
 					className="step-container-v2--goals"
-					topBar={ <Step.TopBar skipButton={ <Step.SkipButton onClick={ handleSkip } /> } /> }
+					topBar={
+						<Step.TopBar
+							skipButton={ <Step.SkipButton onClick={ handleSkip } /> }
+							backButton={
+								site?.plan?.is_free ? (
+									<Step.BackButton
+										onClick={ () => {
+											const destination = addQueryArgs(
+												withLocale( '/setup/onboarding/plans', locale )
```

Comment:
What query args were we intending to add here?

Reply:
I doubt we need the locale segment here. The user is already logged in and it will be ignored in favor of the user's locale.